### PR TITLE
Added documentation for workspaces in O3

### PR DESCRIPTION
**Pull request by vasharma05**
Added documentation for workspaces in O3 as per the latest implementation.

1. Workspaces in O3
2. Implementation
3. Registering workspaces
4. Launching workspaces
5. Siderail nav buttons

### CODE REVIEW

File: pages/docs/workspaces-in-o3/registering-workspaces.mdx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  type: string,
+  canHide?: boolean = false,
+  canMaximize?: boolean = false,
+  width?: 'narrow' | 'wider' = 'narrow',
+  preferredWindowSize?: 'normal' | 'maximized' = 'normal'
+});
+
+```
+
+## Registering workspace as an extension in `routes.json`
```
Comment (ibacher) [inline pages/docs/workspaces-in-o3/registering-workspaces.mdx:41]: This should probably note that it's not the favored way of doing this. Although, and I've been saying this for years now, we should make the workspace into a site-wide concept.

File: pages/docs/workspaces-in-o3/registering-workspaces.mdx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+Workspace have a few properties defining the structure and behaviour of handling the workspace window.
+
+1. **name**: string - Unique identifier for the workspace.
+2. **title**: string - Title shown in the workspace header.
+3. **load**: Function - path to the component to be rendered.
+4. **type**: string - type of the workspace (Default = form)
+5. **canHide**: boolean - can the workspace be collapsed/ hidden (Default = false)
+6. **canMaximize**: boolean - can the workspace be maximized to the full screen(Default = false)
+7. **width**: 'narrow' | 'wider' - Whether the workspace should have a width of 420px (narrow) or 520px (wide, especially for forms with sub forms (sub-sections)) width.
```
Comment (ibacher) [inline pages/docs/workspaces-in-o3/registering-workspaces.mdx:13]:
```suggestion
7. **width**: 'narrow' | 'wider' - Whether the workspace should have a width of 420px (narrow) or 520px (wide, specifically for forms with navigation) width.
```

File: pages/docs/workspaces-in-o3/launching-workspace.mdx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+```ts copy
+import { launchPatientWorkspace } from "@openmrs/esm-patient-common-lib";
+
+const MyComponent() {
+  const openOrderBasket = () => {
+    launchPatientWorkspace(
+      'order-basket',
+      //Optional
+      {
+        title: "Order Basket"
```
Comment (brandones) [inline pages/docs/workspaces-in-o3/launching-workspace.mdx:14]:
```suggestion
        workspaceTitle: "Order basket, courtesy of MyComponent"
```

File: pages/docs/workspaces-in-o3/registering-workspaces.mdx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+7. **width**: 'narrow' | 'wider' - Whether the workspace should have a width of 420px (narrow) or 520px (wider; especially for forms with navigation)
+8. **preferredWindowSize**: 'normal' | 'maximized' - Whether to open the workspace in normal view or in maximized view (Default = 'normal')
+
+> 💡 Please note that `canHide` property should be `true` for workspaces with a `type` that corresponds to a siderail nav button.
+
+Registering a new workspace can be simply done in either of the 2 ways:
+
+## Registering workspace in `index.ts` (Recommended)
+
+A new workspace can be registered in the `index.ts` file, using the `registerWorkspace` function from the `@openmrs/esm-patient-common-lib`
```
Comment (brandones) [inline pages/docs/workspaces-in-o3/registering-workspaces.mdx:20]:
```suggestion
## How to Register a Workspace

A new workspace can be registered in the `index.ts` file, using the `registerWorkspace` function from `@openmrs/esm-patient-common-lib`.
```

File: pages/docs/workspaces-in-o3/registering-workspaces.mdx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+registerWorkspace({
+  name: string,
+  title: string,
+  load: getAsyncLifecycle(() => import(<path-to-component>), options),
+  type: string,
+  canHide?: boolean = false,
+  canMaximize?: boolean = false,
+  width?: 'narrow' | 'wider' = 'narrow',
+  preferredWindowSize?: 'normal' | 'maximized' = 'normal'
```
Comment (brandones) [inline pages/docs/workspaces-in-o3/registering-workspaces.mdx:34]:
```suggestion
  name: "gastrodynamics-form,
  title: translateFrom("@myorg/esm-gastrodynamics-app", "gastrodynamicsTitle", "Gastrodynamics"),
  load: getAsyncLifecycle(() => import("./gastrodynamics.workspace"), options),
  type: "other-form",
  canHide: false,
  canMaximize: true,
  width: 'narrow',
  preferredWindowSize: 'maximized'
```

File: pages/docs/workspaces-in-o3/registering-workspaces.mdx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+});
+
+```
```
Comment (brandones) [inline pages/docs/workspaces-in-o3/registering-workspaces.mdx:37]:
```suggestion

```

You will have to create the file `./gastrodynamics.workspace.tsx` with your workspace component as a default export.
```

File: pages/docs/workspaces-in-o3/registering-workspaces.mdx [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+registerWorkspace({
+  name: string,
+  title: string,
+  load: getAsyncLifecycle(() => import(<path-to-component>), options),
+  type: string,
+  canHide?: boolean = false,
+  canMaximize?: boolean = false,
+  width?: 'narrow' | 'wider' = 'narrow',
+  preferredWindowSize?: 'normal' | 'maximized' = 'normal'
+});
```
Comment (brandones) [inline pages/docs/workspaces-in-o3/registering-workspaces.mdx:35]:
```suggestion

export function startupApp() {
  registerWorkspace({
    name: "gastrodynamics-form,
    title: translateFrom("@myorg/esm-gastrodynamics-app", "gastrodynamicsTitle", "Gastrodynamics"),
    load: getAsyncLifecycle(() => import("./gastrodynamics.workspace"), options),
    type: "other-form",
    canHide: false,
    canMaximize: true,
    width: 'narrow',
    preferredWindowSize: 'maximized'
  });
}
```

**Comment by brandones** [comment]
Okay, rewrote half of this but I need to do something else now. Will try to come back to it. I still don't like that the remaining half is a mix of under-the-hood implementation details that a workspace-maker shouldn't need to know, and actually important information about the interface. There's also no example workspace component, which I think would be a good addition, though that can be a future PR.

**Comment by denniskigen** [comment]
Ping, @brandones, would be great to get this over the line!

**Review by brandones — APPROVED** [review]

**Comment by brandones** [comment]
@denniskigen @ibacher The workspace pages aren't working in the preview? Any idea what's up with that?

File: pages/docs/_meta.en-US.json [hunk 8]
```diff
@@ -11,6 +11,7 @@
   "recipes": "Recipes",
   "extension-system": "Extension system",
   "configuration-system": "Configuration system",
+  "workspaces-in-o3": "Workspaces in O3",
```
Comment (ibacher) [inline pages/docs/_meta.en-US.json:14]:
```suggestion
  "workspaces": "Workspaces in O3",
```

**Comment by ibacher** [comment]
In the `_meta`  file, we referenced the key `workspaces-in-o3`, but the folder is called `workspaces`, so there was no way for the site generator to make the connection between the navigation point and the content. I fixed that above. I don't know exactly why the current iteration is failing, but I imagine the files should have the suffix `.en-US.mdx` because of how the localization works for this static site generator.

**Comment by denniskigen** [comment]
Thanks for the great work, @vasharma05, @brandones !

**Pull request merged**